test/integration/theme-schema.test.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * External dependencies
+ */
+import Ajv from 'ajv';
+
+/**
+ * Internal dependencies
+ */
+import themeSchema from '../../schemas/json/theme.json';
+
+const baseThemeJson = {
+	version: 2,
+};
+
+describe( 'theme.json schema', () => {
+	const ajv = new Ajv( {
+		// Used for matching unknown blocks without repeating core blocks names
+		// in settings.blocks and settings.styles
+		allowMatchingProperties: true,
+	} );
+
+	test( 'schema valid', () => {
+		const result = ajv.validate( themeSchema, baseThemeJson ) || ajv.errors;
```

Comment:
First off, thank you for adding tests here! Based on the number of things you've had to fix, this is sorely needed.

I think we could validate the schema directly rather than indirectly by validating data. I think you should be able to do something like this with AJV.

```suggestion
		const result = ajv.validate( 
			'http://json-schema.org/draft-07/schema#',
			themeSchema
		) || ajv.errors;
```